### ISSUE TITLE
Use of anchors nested in headings causes issues in VoiceOver, promotes anti-pattern usage

### ISSUE THREAD

**Issue by patrickfox** [twbs/bootstrap#16753]
This is a request to fix the accordion example on getbootstrap.com in which headings are used incorrectly to contain anchors. 

Bootstrap accordion example: http://getbootstrap.com/javascript/#collapse-example-accordion
Proposed change: http://jsbin.com/poxowo/edit?html,css,output

### Current markup:

```html
    <div class="panel-heading" role="tab" id="headingOne">
      <h4 class="panel-title">
        <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
          Collapsible Group Item #1
        </a>
      </h4>
    </div>
```

### Issues:

- role=tab should be on the interactive element
- `<button>` is more semantic/appropriate vs. `<a>`, since this anchor doesn't link to anywhere
- anchors within headings(H4) confuse VoiceOver(VO), which:
  - causes VO to place the VO cursor on the H4 vs. the anchor,
  - doesn't read any of the aria attributes on the anchor,
  - breaks the standard VO-Spacebar interaction

This may seem like a VO-specific issue, but the use of interactive elements inside of headings, and the use of anchors vs. buttons in this example promote the use of non-semantic HTML. This example should be updated as follows:

```html
    <div class="panel-heading" id="headingOne">
        <button type="button" data-toggle="collapse" data-parent="#accordion" data-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne" role="tab">
          Collapsible Group Item #1
        </button>
    </div>
```

Thank you!

**Comment by patrickhlauke** [comment]
Agree current accordion implementation isn't completely up to scratch, though some of the issues mentioned are partially VO-specific.

I do not fully agree though that it's not valid or non-semantic to have interactive elements inside headings, if those interactive elements also represent logical headings. And from a practical perspective, an `<a role="button" ...>` (provided it acts the same way) is equivalent to `<button>`, and there are certain accordion approaches (but not in the docs here) which make the link actually explicitly point to the related tab panel.

But in all, consider this put on the to-do list ;)

**Comment by patrickfox** [comment]
@patrickhlauke great - thanks!

**Comment by mdo** [comment]
If this touches any JS in v3, we won't be making any changes for it and will have to revisit in v4 (which currently has no accordion whatsoever).

**Comment by patrickhlauke** [comment]
i tentatively tagges it with JS purely as I'm not sure about how reliant our current code is on the specific structures. Would still somewhere closer to a fix rather than a new feature though, I feel...leaving v3 in a cleaner state.

**Comment by mdo** [comment]
Closing as a won't fix.

The triggers for the accordion are `data` attribute based, so they can be moved a bit in the markup.
